### DISCUSSION TITLE
PLEASE HELP! :( Error: The filename, directory name, or volume label syntax is incorrect while creating C:\Users\Jericho\AppData\Roaming\beets\‪‪

### DISCUSSION THREAD

**Discussion by EchoShauf** [beetbox/beets#5431] [Q&A] [answered]
Everytime I import using "beet import C:\Users\Jericho\Downloads\Music", beets shows the Match, then I proceed to press A to apply, but after that, it keeps giving me an error:

"Error: The filename, directory name, or volume label syntax is incorrect while creating C:\Users\Jericho\AppData\Roaming\beets\  ‪‪D:"

I can't figure out what is wrong, I have been going through the discussions here and straight up even copying people's config.yaml and settings. 

the directory "C:\Users\Jericho\AppData\Roaming\beets\" it points to without the "D:" is where my config.yaml, library and state.pickle is located. 

I thought maybe it's just permissions, then maybe the library: that points to the library.db but somehow it's still not the problem.

Please help me, I have been at it for 3 days now. 
I attached what it looks like on the Powershell. 

Is it just as simple as a directory or invalid filename or character? I cant seem to figure it out please help. 

<img width="800" alt="{70C5E262-A901-4D59-920B-E524658E4587}" src="https://github.com/user-attachments/assets/60a924f6-4c7b-4672-a254-8d5aeab56613">

OS: Windows
Python version: 3.12.6
beets version: 2.0.0

Config.yaml

```
plugins: fetchart embedart info scrub edit fromfilename chroma web permissions missing lastgenre duplicates
directory: ‪‪D:\Beets\Music
library: C:\Users\Jericho\AppData\Roaming\beets\library.db
art_filename: albumart
threaded: yes
original_date: no
per_disc_numbering: no
verbose: 0
asciify_paths: no
max_filename_length: 0

permissions:
    file: 644
    dir: 755

ignore_hidden: yes

paths:
    default: $x__i_artist\$x__i_release_type\$x__i_album%aunique{}\%if{$x__i_multidisc,$x__i_disc\}$x__i_track
    comp: $x__i_artist\$x__i_release_type\$x__i_album%aunique{}\%if{$x__i_multidisc,$x__i_disc\}$x__i_track
    singleton: $x__i_artist\[Non-Album]\%if{$year,[$year]}\$x__i_track
    albumtype_soundtrack: Soundtracks\$album\$track $title
 

import:
    write: yes
    copy: yes
    move: no
    link: no
    delete: no
    resume: ask
    incremental: no
    quiet_fallback: skip
    timid: no
    autotag: yes
    duplicate_action: ask
    
      


lastgenre:
    auto: yes
    source: album

embedart:
    auto: yes

fetchart:
    auto: yes
    cautious: yes
    cover_names: cover front art album folder back
    minwidth: 0
    enforce_ratio: 5%
    sources: filesystem coverart amazon albumart
    google_key: None.
    store_source: no

replaygain:
backend: gstreamer
auto: no

scrub:
    auto: yes

missing:
    count: no
    total: no
    album: no

duplicates:
    album: no
    checksum: ''
    copy: ''
    count: no
    delete: no
    format: ''
    full: no
    keys: []
    merge: no
    move: ''
    path: no
    tiebreak: {}
    strict: no
    tag: ''

pathfields: {}
album_fields: {}

edit:
    albumfields: album albumartist
    itemfields: track title artist album
    ignore_fields: id path

replace:
    '[\\/]': _
    '^\.': _
    '[\x00-\x1f]': _
    '[<>:"\?\*\|]': _
    '\.$': _
    '\s+$': ''
    '^\s+': ''

web:
    host: 0.0.0.0
    port: 8337

**Comment by JOJ0** [comment]
Hi, please put your configuration and code snippets into "code boxes". It makes it easier to read for others and hopefully improves likeliness of help. HTH :-)

**Reply by JOJ0** [reply]
Also, if we talk about configuration, it is critical to see where indentation is because the format in use (YAML) uses indentation for structure. Without proper formatting, we can't check wether your configuration at least "looks" correct.

**Reply by EchoShauf** [reply]
Hi, sorry, I just edited it, please let me know what else I can put here.

**Reply by EchoShauf** [reply]
what's crazy is I had it working before, didn't do anything at all, till it stopped midway, something about queue reaching the limit, so i had to make smaller batches, never touched any pathing or anything.

Then this error now shows as soon as i hit import, it shows me the first option to hit apply, then it gives me the error. 

I just tried it again now and it can't even get past the apply part, as soon as I hit import it just loads then errors.

**Comment by arogl** [comment]
In your paths statements what do they expand to? I cannot see where the below variables are defined in your config

```
$x__i_artist
$x__i_release_type
$x__i_album
```

Can you also add the output of `beet -vv import` for your command?

This is the verbose output to help.

**Reply by EchoShauf** [reply]
Hi, sorry, that was removed, this was the one updated, I posted the wrong one. 

```
    default: $albumartist/$album%aunique{}/$track $title
    singleton: Non-Album/$artist - $title
    comp: Compilations/$album%aunique{}/$track - $title
    albumtype_soundtrack: Soundtracks/$album/$track $title

**Reply by EchoShauf** [reply]
Hi, I tried the verbose -vv and this showed up. 
 
![image](https://github.com/user-attachments/assets/0ad5f8cf-bfa5-4149-9b06-9a4ea2b3e1d6)

**Reply by EchoShauf** [reply]
If i press Y, it would give me errors. 
<img width="853" alt="{6F6ADFF0-74A4-4524-BC04-4277E7D1A895}" src="https://github.com/user-attachments/assets/f9a03bee-972b-4fb3-ab42-3453c73aa1ef">

**Reply by EchoShauf** [reply]
its showing double for the directory when i do the beet config -e

<img width="860" alt="{83E886E9-3055-4057-B71D-7E1C8B620539}" src="https://github.com/user-attachments/assets/104480ac-8494-456f-9ce0-91b77bf9fb37">

**Reply by arogl** [reply, marked as answer]
> If i press Y, it would give me errors. <img alt="{6F6ADFF0-74A4-4524-BC04-4277E7D1A895}" width="853" src="https://private-user-images.githubusercontent.com/161604306/369601895-f9a03bee-972b-4fb3-ab42-3453c73aa1ef.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY5MTM4OTksIm5iZiI6MTcyNjkxMzU5OSwicGF0aCI6Ii8xNjE2MDQzMDYvMzY5NjAxODk1LWY5YTAzYmVlLTk3MmItNGZiMy1hYjQyLTM0NTNjNzNhYTFlZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkyMVQxMDEzMTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03ZjdlMmE3MmE0N2U3MjRhZDcwMzBmYTg4ZTc5MjYwNTJhZDM1YWZmNjAzMmNiYmMyMDdiOTQ1ODZhMDc4NDNlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.wbcglCPXjhNqhvgMrbYiCi8eyqkl5cYVNi0QLQ-QZFc">

When I look at this the directory seems wrong with windows extended characters `\xe2\x80\xaaC`.

[â](https://www.compart.com/en/unicode/U+00e2), [Padding Character](https://www.compart.com/en/unicode/U+0080) and finally  [બ](https://www.compart.com/en/unicode/U+0AAC).

Can you open your config file in an (hex) editor to see if those characters exist in the file?

**Reply by EchoShauf** [reply]
Hello, is notepad ++ a hex editor? Im not really knowledgeable on programming and stuff.

**Reply by EchoShauf** [reply]
Hi, would this be it? I opened it in notepad++ hex editor.


<img width="495" alt="image" src="https://github.com/user-attachments/assets/e09aaf44-ae71-48b6-a47d-12c69fcb4250">

**Reply by EchoShauf** [reply]
HOLY MOLY! I would have never figured it out.! THANK YOU. I tried to rewrite the entire directory and library path and it didnt give me the error anymore. omg

**Reply by EchoShauf** [reply]
<img width="868" alt="{0FA321AC-7EA9-499C-93F2-CC6218ADF7E2}" src="https://github.com/user-attachments/assets/ba5cef11-476a-46e0-b56a-32a2ea303c18">
